bokeh/charts/builder/boxplot_builder.py

```diff
@@ -33,6 +33,43 @@
 
 def BoxPlot(values, marker="circle", outliers=True, xscale="categorical", yscale="linear",
         xgrid=False, ygrid=True, **kw):
+    """ The `BoxPlot` chart creates a BoxPlot chart using the `BoxPlotBuilder` to
+     render the geometry from `values`, the `marker` and `outliers` arguments.
+
+     Args:
+            values (iterable): iterable 2d representing the data series
+                values matrix.
+            marker (int or string, optional): if outliers=True, the marker type to use
+                e.g., `circle`.
+            outliers (bool, optional): Whether or not to plot outliers.
+
+    In addition the the parameters specific to this chart,
+    :ref:`charts_generic_arguments` and :ref:`chart_builders_generic_arguments`
+    are also accepted as keyword parameters.
+
+    .. bokeh-plot::
+        :source-position: above
+
+        from collections import OrderedDict
+        from bokeh.charts import BoxPlot
+        import numpy as np
+        medals = OrderedDict([
+                    ('bronze', np.array([7.0, 10.0, 8.0, 7.0, 4.0, 4.0, 1.0, 5.0, 2.0, 1.0,
+                                4.0, 2.0, 1.0, 2.0, 4.0, 1.0, 0.0, 1.0, 1.0, 2.0,
+                                0.0, 1.0, 0.0, 0.0, 1.0, 1.0])),
+                    ('silver', np.array([8., 4., 6., 4., 8., 3., 3., 2., 5., 6.,
+                                1., 4., 2., 3., 2., 0., 0., 1., 2., 1.,
+                                3.,  0.,  0.,  1.,  0.,  0.])),
+                    ('gold', np.array([6., 6., 6., 8., 4., 8., 6., 3., 2., 2.,  2.,  1.,
```

Comment:
> Do the values need to be np.array? could they just be python lists? If not we should just go whole-hog and use pandas here maybe.

Unfortunately yes... :-(  #1876 

> Also I think I like the way OrderedDict is constructed more in the other example below. (i.e., Medals = OrderedDict() then adding each key)
I am little worried about using OrderedDict so much I don't actually think it is what would be common with actual users.

Yeap. I prefer the example below as well. I forgot to expand the definition after copy and paste..

Apart from that, I've used OrderedDict because I've took the inputs from tests and OD was used there to maintain series order (useful on tests). So.. maybe we could mix all supported inputs on different builders to highlight all the different input types?